diff --git a/src/execution/WorkflowExecutionCoordinator.ts b/src/execution/WorkflowExecutionCoordinator.ts
index b781355..61f75b2 100644
--- a/src/execution/WorkflowExecutionCoordinator.ts
+++ b/src/execution/WorkflowExecutionCoordinator.ts
@@ -13,10 +13,17 @@ export class WorkflowExecutionCoordinator {
     delay: number,
     callback: () => void
   ): void {

    if (
      this.lifecycle.getState(workflow.id) === "archived"
    ) {
      return;
    }

     this.scheduler.schedule(
       workflow,
       delay,
       callback
     );
   }
 }
